src/components/ModalWindowCheckout/ModalWindowCheckout.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC, useState } from 'react';
+import styles from './styles.module.scss';
+import { PrimaryButton } from '../PrimaryButton';
+import { PrimaryButtons } from '@/enums/PrimaryButtons';
+import { useRemoveFromCartButton } from '@/hooks';
+import { useSelector } from 'react-redux';
+import { RootState } from '@/app/store';
+import { getCartProducts, getTotalCost } from '@/utils';
+import { useNavigate } from 'react-router-dom';
+import { Paths } from '@/enums';
+import { ModalWindowSuccess } from '../ModalWindowSuccess/ModalWindowSuccess';
+
+interface PropsModalWindow {
+  isOpen: boolean;
+  onClickCancel: () => void;
+  onCartEmpty: () => void;
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const ModalWindowCheckout: FC<PropsModalWindow> = ({
+  isOpen,
+  onClickCancel,
+  setIsModalOpen,
+  onCartEmpty,
+}) => {
+  const cart = useSelector((state: RootState) => state.cart);
+  const cartItems = getCartProducts(cart.items);
+  const totalAmount = getTotalCost(cartItems);
+
+  const handleRemove = useRemoveFromCartButton();
+  const navigate = useNavigate();
+
+  const [isSuccessModalOpen, setIsSuccessModalOpen] = useState(false);
+
+  const handleRemoveOnConfirm = () => {
+    setIsModalOpen(false);
+    setIsSuccessModalOpen(true);
+
+    setTimeout(() => {
+      navigate(Paths.HOME);
+      onCartEmpty();
+    }, 3000);
+
+    cartItems.forEach(cartItem => {
+      handleRemove(cartItem.id);
+    });
+  };
+
+  return (
+    <>
+      {isOpen && (
+        <div className={styles.modal}>
+          <div className={styles.modal__content}>
+            <h2 className={styles.modal__title}>Complete your purchase</h2>
+
+            <div className={styles.modal__tableContainer}>
+              <table className={styles.modal__table}>
+                <thead>
+                  <tr>
+                    <th>Product</th>
+                    <th>Quantity</th>
+                    <th>Price</th>
```

Comment:
I prefer the idea of putting the headers of all sorts of tables into the enum and going through it to render them, but it's not critical